app/api/notifications.py

```
@@ -0,0 +1,517 @@
+    
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to create notification: {str(e)}"
+        )
+
+
+@router.post("/bulk", response_model=List[NotificationResponse])
+async def create_bulk_notifications(
+    notification_data: NotificationBulkCreate,
+    db: AsyncSession = Depends(get_db),
```

Ruff (0.13.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)